types/d3-axis/index.d.ts

```diff
@@ -29,53 +29,53 @@ export interface AxisTimeInterval {
 
 /**
  * A helper interface to which a scale passed into axis must conform (at a minimum)
- * for axis to use the scale without error
+ * for axis to use the scale without error.
  */
 export interface AxisScale<Domain> {
     (x: Domain): number | undefined;
     domain(): Domain[];
     range(): number[];
     copy(): this;
     bandwidth?(): number;
-    // TODO: Reconsider the below, note that the compiler does not  differentiate the overloads w.r.t. optionality
+    // TODO: Reconsider the below, note that the compiler does not differentiate the overloads w.r.t. optionality
     // ticks?(count?: number): Domain[];
     // ticks?(count?: AxisTimeInterval): Date[];
     // tickFormat?(count?: number, specifier?: string): ((d: number) => string);
     // tickFormat?(count?: number | AxisTimeInterval, specifier?: string): ((d: Date) => string);
 }
 
 /**
- * A helper type to alias elements which can serve as a container for an axis
+ * A helper type to alias elements which can serve as a container for an axis.
  */
 export type AxisContainerElement = SVGSVGElement | SVGGElement;
 
 /**
- * Interface defining an axis generator. The generic <Domain> is the type of the axis domain
+ * Interface defining an axis generator. The generic <Domain> is the type of the axis domain.
  */
 export interface Axis<Domain> {
```

Comment:
I will review the constraints on `Domain` in the definition for `Axis` and `AxisScale` to define a suitable default for the generic parameters (TS 2.3 feature that we should use here as well). We might be able to do better than `any` for the most general permissible case.

We could have `export interface Axis<Domain extends DomainConstraint = DomainDefault> {...}` and `export interface AxisScale<Domain extends DomainConstraint = DomainDefault> {...}`. Then in day-to-day use one could just do `let x: Axis;` for brevity, if the defaults are good enough.